integrations/docker/images/chip-build/Dockerfile

```diff
@@ -27,14 +27,16 @@ RUN set -x \
     libcairo2-dev \
     libdbus-1-dev \
     libdbus-glib-1-dev \
+    libdmalloc-dev \
     libgif-dev \
     libglib2.0-dev \
     libical-dev \
     libjpeg-dev \
-    libdmalloc-dev \
     libmbedtls-dev \
     libncurses5-dev \
     libncursesw5-dev \
+    libnl-3-dev \
+    libnl-route-3-dev \
```

Comment:
Is this used statically? You may need this in all other Dockerfiles that build otherwise

Reply:
For building, all other dockerfiles will inherit from this and will just work.
Cirque also inherits from this, so unit tests will pass.

Execution and cross-compilation will likely need separate consideration. For cross compile, we either disable this or add the library to the sysroot. the sysroot is built outside of github so will not show up as a patch.

Reply:
Why not change the default to avahi?

The goal of minimal mDNS was to be minimal and available without system deps, dependency creep here is quite undesirable. You can swap out the dependency on libnl with one on avahi-client.

Reply:
It's going to be a lot of work to catch up with a standalone mDNS resolver in capabilities.

Likely minimal mDNS should not be used in production Linux deployment, at all. So I don't understand this direction.

Reply:
This is a system dependency rather than minmdns. Minmdns is affected because the underlying inet layer reports "here are ip addresses you can use" when in fact they are not as usable in general.

As an example, chromium https://chromium.googlesource.com/chromium/src/net/+/refs/heads/main/base/network_interfaces_linux.cc#57 also does  various checks to ensure that applications are only presented usable things. I think it is a failure of our platform/inet that we don't have such capabilities.

So I would like to have an option to potentially add such tests, even if behind a flag.

Regarding usage afterwards, we have to determine if we really want inside inet/platform (which adds a dependency) or have an example on how to override minmdns policies regarding addresses (need to design an API for that)